node/src/components/storage.rs

```diff
@@ -1035,50 +1130,44 @@ impl Storage {
         &self,
         tx: &mut Tx,
         era_id: EraId,
-    ) -> Result<Option<Block>, LmdbExtError> {
+    ) -> Result<Option<Block>, Error> {
         self.switch_block_era_id_index
             .get(&era_id)
             .and_then(|block_hash| self.get_single_block(tx, block_hash).transpose())
             .transpose()
     }
 
     /// Retrieves the state root hashes from storage to check the integrity of the trie store.
-    pub fn read_state_root_hashes_for_trie_check(&self) -> Option<Vec<Blake2bHash>> {
+    pub fn read_state_root_hashes_for_trie_check(&self) -> Result<Vec<Blake2bHash>, Error> {
         let mut blake_hashes: Vec<Blake2bHash> = Vec::new();
-        let txn =
-            self.env.begin_ro_txn().ok().unwrap_or_else(|| {
-                panic!("could not open storage transaction for trie store check")
-            });
-        let mut cursor = txn
-            .open_ro_cursor(self.block_header_db)
-            .ok()
-            .unwrap_or_else(|| panic!("could not create cursor for trie store check"));
+        let txn = self.env.begin_ro_txn()?;
+        let mut cursor = txn.open_ro_cursor(self.block_header_db)?;
         for (_, raw_val) in cursor.iter() {
-            let header: BlockHeader = lmdb_ext::deserialize(raw_val).ok()?;
+            let header: BlockHeader = lmdb_ext::deserialize(raw_val)?;
             let blake_hash = Blake2bHash::from(*header.state_root_hash());
             blake_hashes.push(blake_hash);
         }
 
         blake_hashes.sort();
         blake_hashes.dedup();
 
-        Some(blake_hashes)
+        Ok(blake_hashes)
     }
 
     /// Retrieves a single block header in a separate transaction from storage.
     fn get_single_block_header<Tx: Transaction>(
         &self,
         tx: &mut Tx,
         block_hash: &BlockHash,
-    ) -> Result<Option<BlockHeader>, LmdbExtError> {
+    ) -> Result<Option<BlockHeader>, Error> {
         let block_header: BlockHeader = match tx.get_value(self.block_header_db, &block_hash)? {
             Some(block_header) => block_header,
             None => return Ok(None),
         };
         let found_block_header_hash = block_header.hash();
         if found_block_header_hash != *block_hash {
-            return Err(LmdbExtError::BlockHeaderNotStoredUnderItsHash {
-                queried_block_hash: *block_hash,
+            return Err(Error::BlockHeaderNotStoredUnderItsHash {
+                queried_block_hash_bytes: Digest::from(*block_hash).to_vec(),
```

Comment:
Why the conversion? Isn't `found_block_header_hash` also a `BlockHash`?

Reply:
Elsewhere, this same error is used when reading values under a raw key in the block header DB using a cursor.  This is why it has this structure.

I've switched to `block_hash.as_ref().to_vec()` in https://github.com/casper-network/casper-node/pull/1711/commits/c3982d4ac73d6e54481bfd4691f52214b059b99c